### ISSUE TITLE
Marathon not accepting implicit roles

### ISSUE THREAD

**Issue by Spritekin** [mesosphere/marathon#3854]
I'm running on mesos 0.28, Marathon 1.1.1.

My slaves define a "prometheus" resource role.
My mesos-master does not define the "prometheus" resource role because on mesos 0.28 it is not required. In fact the flag is deprecated.

If marathon does not define the mesos-role, then it won't get "prometheus" resources from mesos.
If marathon defines the "prometheus" mesos-role then it throws 
WARN Error: Role 'prometheus' is not present in the master's --roles

So in other words am I'm forced to define the deprecated --roles even when deprecated?

Edit: I just noticed we never got rid of the roles old config. This means the master is still using the --roles and the prometheus role was not there. 

So I will test without the roles and report back if necessary. Sorry for this.
